### PR TITLE
Fix KUBERNETES_SERVICE_ACCOUNT_TOKEN auth

### PR DESCRIPTION
**Pull request by thesuperzapper**
### What changes were proposed in this pull request?

This PR fixes the `KUBERNETES_SERVICE_ACCOUNT_TOKEN` auth type, as it currently will never work.

The current problem is caused by a VERY bad assumption being made by Kubeflow itself. Inside the `kfp.Client()` initialization code, if a `host` is specified it will never enable ServiceAccount auth. ([Relevant code in `kfp.Client()`](https://github.com/kubeflow/pipelines/blob/1e032f550ce23cd40bfb6827b995248537b07d08/sdk/python/kfp/_client.py#L269-L271))

Because we require that `api_endpoint` is set and [pass it to the `kfp.Client()` constructor](https://github.com/elyra-ai/elyra/blob/741440086b7179c1fb14daf1f7a6a3a13777ac22/elyra/pipeline/kfp/processor_kfp.py#L135), we always specify a `host`.

Luckily the solution is relatively simple, we can explicitly tell the `kfp.Client()` constructor that we want to use ServiceAccount auth, by passing a `kfp.auth.ServiceAccountTokenVolumeCredentials()` to the `credentials` argument of `kfp.Client()`.

### How was this pull request tested?

- Running Elyra in a Kubeflow cluster with ServiceAcount auth

----
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ptitzler** [comment]
@alexisc67 sounds like the problem you've brought up on gitter!

File: elyra/pipeline/kfp/kfp_authentication.py [hunk 1]
```diff
@@ -628,7 +633,7 @@ class K8sServiceAccountTokenAuthenticator(AbstractAuthenticator):
 
     def authenticate(self,
                      kf_endpoint: str,
-                     runtime_config_name: str) -> None:
+                     runtime_config_name: str) -> TokenCredentialsBase:
```
Comment (ptitzler) [inline elyra/pipeline/kfp/kfp_authentication.py:636]: Can you please also update the doc string (`:return: ...`) to reflect the change?

**Review by ptitzler — APPROVED** [review]
LGTM

File: elyra/pipeline/kfp/kfp_authentication.py (hunk 1 above)
Comment (thesuperzapper) [reply elyra/pipeline/kfp/kfp_authentication.py]: Fixed in https://github.com/elyra-ai/elyra/pull/2358/commits/0c354d5ba8fe7c31026f9a11f6a0a3d0a6ab42a0

**Comment by thesuperzapper** [comment]
@ptitzler this PR should be ready to merge now.

**Pull request merged**